bluepyemodel/emodel_pipeline/plotting.py

```diff
@@ -1255,6 +1188,515 @@ def run_and_plot_EPSP(
         )
 
 
+def plot_IV_curves(
+    evaluator, emodels, figures_dir, efel_settings, prot_name="iv", write_fig=True, n_bin=5
+):
+    """Plots IV curves of peak voltage and voltage_deflection
+    with simulated and experimental values. Only works for threshold-based protocols.
+
+    Args:
+        evaluator (CellEvaluator): cell evaluator
+        emodels (list): list of EModels
+        figures_dir (str or Path): output directory for the figure to be saved on
+        efel_settings (dict): eFEL settings in the form {setting_name: setting_value}.
+        prot_name (str): name of protocol of which recordings to use
+        write_fig (bool): whether to save the figure
+        n_bin (int): number of bins to use
+    """
+    # pylint: disable=too-many-nested-blocks, possibly-used-before-assignment
+    # note: should maybe also check location and recorded variable
+    make_dir(figures_dir)
+    if efel_settings is None:
+        efel_settings = bluepyefe.tools.DEFAULT_EFEL_SETTINGS.copy()
+
+    # add missing features (if any) to evaluator
+    updated_evaluator = fill_in_IV_curve_evaluator(evaluator, efel_settings, prot_name)
+
+    emodel_name = None
+    cells = None
+    for emodel in emodels:
+        # -- get extracted IV curve data -- #
+        # do not re-extract data if the emodel is the same as previously
```

Comment:
Is there a reason to only check for the last processed?

Reply:
Because then we don't have to re-extract data to refill our variables with exactly the same values. I am not keeping every instance of extracted data because: it would take memory, and I don't expect `emodels` list to contain one emodel, then another one, then the 1st one again for most use cases